src/camera.cc

```diff
@@ -233,11 +268,11 @@ bool Camera::CastShadowRay(Ray& ray, Scene& scene, Direction& light_direction) {
 
 ColorDbl Camera::Raytrace(Ray& ray, Scene& scene, unsigned int depth) {
   float z_buffer = FLT_MAX;
-  IntersectionPoint* intersection_point = GetClosestIntersectionPoint(ray, scene);
-  if (intersection_point && depth < MAX_DEPTH) {
+  std::unique_ptr<IntersectionPoint> intersection_point = GetClosestIntersectionPoint(ray, scene);
+  if (intersection_point) {
```

Comment:
First thing in Shade. Instead of just returning black, return local lightning model